### PR TITLE
[Keyboard] Add "craftWalk" keyboard

### PR DESCRIPTION
**Pull request by sotoba**
## Description

Add craftWalk keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

none

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/craftwalk/config.h [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#ifndef LINK_TIME_OPTIMIZATION_ENABLE
+  #define NO_ACTION_MACRO
+  #define NO_ACTION_FUNCTION
+#endif
```
Comment (zvecr) [inline keyboards/craftwalk/config.h:198]:
```suggestion
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
```

File: keyboards/craftwalk/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 sotoba
```
Comment (zvecr) [inline keyboards/craftwalk/keymaps/default/config.h:1]: This file can be deleted

File: keyboards/craftwalk/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# craftWalk
+
+<img src="https://github.com/sotoba/craftwalk/blob/images/craftwalk_main.jpg" width="640">
```
Comment (zvecr) [inline keyboards/craftwalk/readme.md:3]: dead image link

File: keyboards/craftwalk/rules.mk [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (zvecr) [inline keyboards/craftwalk/rules.mk:23]:
```suggestion
```

File: keyboards/craftwalk/readme.md (hunk 3 above)
Comment (sotoba) [reply keyboards/craftwalk/readme.md:3]: oops
target repository was private, modified setting to public

File: keyboards/craftwalk/keymaps/default/config.h (hunk 2 above)
Comment (sotoba) [reply keyboards/craftwalk/keymaps/default/config.h]: deleted it

File: keyboards/craftwalk/info.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+              {"label":"Mouse", "x":2.25, "y":2},
+              {"label":"F", "x":4.25, "y":2},
+              {"label":"Adjust", "x":1.25, "y":2.25},
+              {"label":"Mouse", "x":3.25, "y":2.25},
+              {"label":"Num", "x":5.25, "y":3, "h":1.5},
+              {"label":"Space", "x":6.25, "y":3, "h":1.5}
+            ]
+        }
+    }
+}
```
Comment (zvecr) [inline keyboards/craftwalk/info.json:27]:
```suggestion
{
    "keyboard_name": "craftwalk",
    "url": "https://github.com/sotoba/craftwalk",
    "maintainer": "sotoba",
    "width": 7.25,
    "height": 4,
    "layouts": {
        "LAYOUT": {
            "layout": [
              {"label":"Q", "x":1.25, "y":0.25},
              {"label":"W", "x":2.25, "y":0},
              {"label":"E", "x":3.25, "y":0.25},
              {"label":"Ctrl", "x":0, "y":1},
              {"label":"A", "x":1.25, "y":1.25},
              {"label":"S", "x":2.25, "y":1},
              {"label":"D", "x":3.25, "y":1.25},
              {"label":"Shift", "x":0, "y":2},
              {"label":"Adjust", "x":1.25, "y":2.25},
              {"label":"Mouse", "x":2.25, "y":2},
              {"label":"Mouse", "x":3.25, "y":2.25},
              {"label":"F", "x":4.25, "y":2},
              {"label":"Num", "x":5.25, "y":3, "h":1.5},
              {"label":"Space", "x":6.25, "y":3, "h":1.5}
            ]
        }
    }
}
```

File: keyboards/craftwalk/config.h [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
```
Comment (fauxpark) [inline keyboards/craftwalk/config.h:247]:
```suggestion
```

File: keyboards/craftwalk/rules.mk [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/craftwalk/rules.mk:31]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```

File: keyboards/craftwalk/rules.mk [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+# Build Options
```
Comment (fauxpark) [inline keyboards/craftwalk/rules.mk:13]:
```suggestion

# Build Options
```

**Comment by sotoba** [comment]
Reflected all suggested changes. Does the review take time?

File: keyboards/craftwalk/rules.mk [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/craftwalk/rules.mk:31]:
```suggestion
```

File: keyboards/craftwalk/rules.mk [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+EXTRAKEY_ENABLE = no        # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
```
Comment (fauxpark) [inline keyboards/craftwalk/rules.mk:28]:
```suggestion
```

File: keyboards/craftwalk/rules.mk [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/craftwalk/rules.mk:11]:
```suggestion
```

**Comment by sotoba** [comment]
One week has passed since I opened this PR. @zvecr, are there any issues to approve?

File: keyboards/craftwalk/info.json [hunk 12]
```diff
@@ -0,0 +1,27 @@
+{
+    "keyboard_name": "craftwalk",
+    "url": "https://github.com/sotoba/craftwalk",
+    "maintainer": "sotoba",
+    "width": 7.25,
+    "height": 4,
```
Comment (noroadsleft) [inline keyboards/craftwalk/info.json:6]:
```suggestion
    "height": 4.5,
```

File: keyboards/craftwalk/config.h [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x2E8F
+#define DEVICE_VER 0x0001
+#define MANUFACTURER sotoba
+#define PRODUCT craftwalk
+#define DESCRIPTION A macro pad for (mine)crafters
```
Comment (fauxpark) [inline keyboards/craftwalk/config.h:28]:
```suggestion
```

File: keyboards/craftwalk/config.h [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (fauxpark) [inline keyboards/craftwalk/config.h:219]:
```suggestion

```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**